tensorboard/plugins/profile/profile_plugin.py

```diff
@@ -327,12 +329,32 @@ def data_route(self, request):
       return http_util.Respond(request, '404 Not Found', 'text/plain', code=404)
     return http_util.Respond(request, data, 'application/json')
 
+  @wrappers.Request.application
+  def capture_route(self, request):
+    service_addr = request.args.get('service_addr')
+    duration = int(request.args.get('duration'))
+    is_tpu_name = request.args.get('is_tpu_name') == 'true'
+    if is_tpu_name:
+      tpu_cluster_resolver = (
+        tf.contrib.cluster_resolver.TPUClusterResolver([service_addr]))
+      service_addr = tpu_cluster_resolver.get_master()
+    service_addr = service_addr.replace('grpc://', '').replace(':8470', ':8466')
```

Comment:
Sorry for my ignorance but (1) where does this port number come from? (2) can we use urlparse to strip out the scheme or at least match with regex like `^grpc://`?

Reply:
I am not sure this part as well. Copy from capture_tpu_profile. +qiuminxu

Reply:
The port number is fixed by the tpu tfrc runtime. The 8470 is the port to run a TPU model, while the 8466 is the port to tpu profiler service. Match with regex is probably cleaner.

Reply:
Moved it under condition is_tpu_name and added comments.